1-Draft/RFCNNNN-VSCode-PSScriptAnalyzer-Rules.md

```diff
@@ -0,0 +1,151 @@
+---
+RFC: RFC<four digit unique incrementing number assigned by Committee, this shall be left blank by the author>
+Author: Tyler Leonhardt
+Status: Draft
+SupercededBy: N/A
+Version: 0.1
+Area: Tooling - VSCode extension & PSScriptAnalyzer
+Comments Due: December 15th
+Plan to implement: Yes
+---
+
+# New default rules in the VSCode Extension and a new "Aggressive Analysis mode" rules
+
+## Context
+The VSCode extension provides an integration with PSScriptAnalyzer (referred to as PSSA going forward) to expose rule violations in a friendly way.
+These rule violations show up as green or red squiggles under the associated code that violates the rules.
+Here's an example:
+
+![example hover dialog](https://i.imgur.com/ulR5Bhv.png)
+
+Today,
+the VSCode extension only enables a subset of the default PSSA rules.
+This was done because some of the PSSA default rules can be opinionated and we don't want to intrude on user's that don't want too much chatter.
+If a user wanted to opt-in for more rules,
+they have a few options:
+
+- Using the
+"PowerShell: Select PSScriptAnalyzer rules"
+command pallet action
+- Create a
+`PSScriptAnalyzerSettings.psd1`
+file with the rules you want and update your
+`powershell.scriptAnalysis.settingsPath`
+VSCode setting to point to that
+
+## Problem
+
+Using these mechanisms require understanding of what PSSA is,
+what rules are,
+how to configure PSSA,
+etc.
+
+It's a lot of information to go digging for.
+
+It should be easier to opt-in to more rules without needing to go through PSSA docs.
+
+## Solution - "Aggressive Analysis mode"
+
+I'm proposing an 
+"Aggressive Analysis mode"
+set of rules that you can flip on with one setting in VSCode. 
+In addition,
+adding a few other rules to the default rules.
+
+`powershell.scriptAnalysis.mode`
+
+This can be set to:
+
+- None - disables script analysis
+- Default - some crutial rules but nothing too crazy
+- Aggressive - more rules, more fun
+
+Here is the table of:
+
+- the current Defaults
+- the ones that will be a part of the "Aggressive analysis mode"
+- the ones we are considering to add to the Defaults (marked with an x)
+
+| Rule                                         | PSSA Type   | Mode       | Add to default |
+|----------------------------------------------|-------------|------------|----------------|
+| AlignAssignmentStatement                     | Warning     |            |                |
+| AvoidAssignmentToAutomaticVariable           | Warning     | Aggressive |                |
```

Comment:
I would mark this as default because the rule was written very mildly to only warn about read-only automatic variables that will 100% cause a runtime error.